### ISSUE TITLE
`D400`/`D415` false positive for `click` command docstrings

### ISSUE THREAD

**Issue by olivierdelree** [astral-sh/ruff#23876]
### Summary

When documenting [click](https://click.palletsprojects.com/en/stable/) commands, it is possible to [truncate help text](https://click.palletsprojects.com/en/stable/documentation/#truncating-help-texts) by using `\f` in the docstring.

I often run into [D400 (missing-trailing-period)](https://docs.astral.sh/ruff/rules/missing-trailing-period/)/[D415 (missing-terminal-punctuation)](https://docs.astral.sh/ruff/rules/missing-terminal-punctuation/) when only the first line should be used for the command documentation. In such cases, Ruff reports that the first line does not end in a period when it does.

```python
import sys

import click


@click.command()
@click.option("-b", "--bar", type=int, help="A description for the use of bar.")
def foo(bar: int | None = None) -> None:
    """Lorem ipsum.
    \f

    Args:
        bar: This is documentation relevant to using foo from Python directly.
    """
    print(bar)


if __name__ == "__main__":
    foo(sys.argv[1:])
```
```bash
$ uv run ruff check --select D400 main.py
D400 First line should end with a period
  --> main.py:9:5
   |
 7 |   @click.option("-b", "--bar", type=int, help="A description for the use of bar.")
 8 |   def foo(bar: int | None = None) -> None:
 9 | /     """Lorem ipsum.
10 | |     \f
11 | |
12 | |     Args:
13 | |         bar: This is documentation relevant to using foo from Python directly.
14 | |     """
   | |_______^
15 |       print(bar)
   |
help: Add period

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

Putting a period after the `\f` (on the same line) makes the error go away but then documentation ends up with a random period.

**Comment by ntBre** [comment]
This is very closely related to https://github.com/astral-sh/ruff/issues/23878. Hopefully we can adjust the behavior for all of the `D` rules at once if others are supportive of that idea.
